src/pages/item/item-actions/ItemActions.tsx

```diff
@@ -1,63 +1,169 @@
-import React, { useState } from 'react';
-import { useGetItemRelationsQuery } from '../../../api/codeBeamerApi';
-import Importer from '../../import/components/importer/Importer';
-
-export default function ItemActions(props: { itemId: string | number }) {
-	const [disabled, setDisabled] = useState<boolean>(true);
-	const [itemIds, setItemIds] = useState<string[]>([]);
-	const [queryString, setQueryString] = useState<string>('');
-
-	const { data, error, isLoading } = useGetItemRelationsQuery(props.itemId);
-
-	const clickHandler = () => {
-		if (data && !disabled) {
-			const ids = data.downstreamReferences.map((d) =>
-				d.itemRevision.id.toString()
-			);
-			setItemIds(ids);
-			setQueryString(`item.id IN (${ids.join(',')})`);
-		} else {
-			console.warn(
-				"Can't load Downstream References - data still loading or failed to do so."
-			);
-		}
-	};
-
-	React.useEffect(() => {
-		if (error) {
-			setDisabled(true);
-			return;
-		} else if (data) {
-			if (!data.downstreamReferences.length) {
-				setDisabled(true);
-			} else {
-				setDisabled(false);
-			}
-		}
-	}, [data, error]);
-
-	return (
-		<div className="flex-centered">
-			<button
-				className={`button button-tertiary ${
-					isLoading ? 'button-loading button-loading-primary' : ''
-				}`}
-				onClick={clickHandler}
-				disabled={disabled}
-				data-test="load-downstream-references"
-				title="Load the Item's Downstream References onto the board, if they're not yet there"
-			>
-				{!isLoading && (
-					<>
-						<span className="icon-add-row-bottom"></span>
-						<span>Load Downstream References</span>
-					</>
-				)}
-				{data && ` (${data.downstreamReferences.length})`}
-			</button>
-			{queryString && (
-				<Importer items={itemIds} queryString={queryString} />
-			)}
-		</div>
-	);
+import React, { useState } from "react";
+import { useGetItemRelationsQuery } from "../../../api/codeBeamerApi";
+import { createConnectorsForDownstreamRefsAndAssociation } from "../../../api/miro.api";
+import Importer from "../../import/components/importer/Importer";
+import getAppCardId from "../../../api/utils/getAppCardId";
+
+export default function ItemActions(props: {
+  itemId: string | number;
+  cardId: string | number;
+}) {
+  const [
+    loadDownstreamReferencesDisabled,
+    setloadDownstreamReferencesDisabled,
+  ] = useState<boolean>(true);
+  const [showDependenciesDisabled, setShowDependenciesDisabled] =
+    useState<boolean>(true);
+  const [itemIds, setItemIds] = useState<string[]>([]);
+  const [queryString, setQueryString] = useState<string>("");
+  const [downstreamRefs, setDownstreamRefs] = useState<number[]>([]);
+  const [associations, setAssociations] = useState([]);
```

Comment:
I don't know what your compiler thinks of this, but mine certainly complains loudly that this state variable here doesn't have a type..